couchbase-ns/src/main/java/com/excelian/mache/couchbase/CouchbaseConfig.java

```diff
@@ -1,170 +1,170 @@
-            this.adminPassword = adminPassword;
-            return this;
-        }
-
-        public Builder withServerAdresses(List<String> serverAddresses) {
-            this.serverAddresses = serverAddresses;
-            return this;
-        }
-
-        public Builder withBucketName(String bucketName) {
-            this.bucketName = bucketName;
-            return this;
-        }
-
-        public Builder withBucketPassword(String bucketPassword) {
-            this.bucketPassword = bucketPassword;
-            return this;
-        }
-
-        public Builder withBucketSize(int bucketSize) {
-            this.bucketSize = bucketSize;
-            return this;
-        }
-
-        public Builder withNumReplicas(int numReplicas) {
-            this.numReplicas = numReplicas;
-            return this;
-        }
-
-        public Builder withFlushEnabled(boolean flushEnabled) {
-            this.flushEnabled = flushEnabled;
-            return this;
-        }
-
-        public Builder withSchemaOptions(SchemaOptions schemaOptions) {
-            this.schemaOptions = schemaOptions;
-            return this;
-        }
-
-        public Builder withCouchbaseEnvironment(CouchbaseEnvironment couchbaseEnvironment) {
-            this.couchbaseEnvironment = couchbaseEnvironment;
-            return this;
-        }
-
-        public CouchbaseConfig build() {
-            if (bucketName == null || bucketName.isEmpty()) {
-                throw new IllegalArgumentException("Must provide bucket name.");
-            } else if (cacheType == null) {
-                throw new IllegalArgumentException("Must provide cache type.");
-            }
-
-            return new CouchbaseConfig(cacheType, couchbaseEnvironment, adminUser, adminPassword, serverAddresses,
-                    bucketName, bucketPassword, bucketSize, numReplicas, flushEnabled, schemaOptions);
-        }
-    }
-
-
-    public static CouchbaseConfig.Builder builder() {
-        return new Builder();
-    }
-
-
-    @SuppressWarnings("unchecked")
-    public <V> Class<V> getCacheType() {
-        return cacheType;
-    }
-
-    public String getAdminUser() {
-        return adminUser;
-    }
-
-    public String getAdminPassword() {
-        return adminPassword;
-    }
-
-    public List<String> getServerAddresses() {
-        return serverAddresses;
-    }
-
-    public String getBucketName() {
-        return bucketName;
-    }
-
-    public String getBucketPassword() {
-        return bucketPassword;
-    }
-
-    public int getBucketSize() {
-        return bucketSize;
-    }
-
-    public int getNumReplicas() {
-        return numReplicas;
-    }
-
-    public boolean isFlushEnabled() {
-        return flushEnabled;
-    }
-
-    public SchemaOptions getSchemaOptions() {
-        return schemaOptions;
-    }
-
-    public CouchbaseEnvironment getCouchbaseEnvironment() {
-        return couchbaseEnvironment;
-    }
-}
+package com.excelian.mache.couchbase;
```

Comment:
We appear to be using the lambda builder pattern for the Cassandra and Mongo builder classes, is this something we will want to use here?